test: ajoute des tests unitaires pour les DAO Hibernate

Odonata971 opened the pull request:
Ajoute des classes de test unitaires pour améliorer la couverture du code des DAO Hibernate abstraits en utilisant des mocks pour les interactions avec l'EntityManager.

Vérifie le bon fonctionnement des méthodes de création, lecture, mise à jour, et suppression des entités de test.

Garantit la robustesse des DAO en simulant différents scénarios de recherche et suppression avec des entités inexistantes.

Commit:
test: ajoute des tests unitaires pour les DAO Hibernate

Ajoute des classes de test unitaires pour améliorer la couverture
du code des DAO Hibernate abstraits en utilisant des mocks pour
les interactions avec l'EntityManager.

Vérifie le bon fonctionnement des méthodes de création, lecture,
mise à jour, et suppression des entités de test.

Garantit la robustesse des DAO en simulant différents scénarios de
recherche et suppression avec des entités inexistantes.

Francois389 commented on src/test/java/sae/semestre/six/generic/AbstractHibernateDaoTest.java:
```diff
@@ -0,0 +1,171 @@
+
+
+/**
+ * Cette classe teste la classe {@link AbstractHibernateDao} et {@link GenericDao}, la dernière classe citée
```
In English, please.

Francois389 commented on src/test/java/sae/semestre/six/generic/AbstractHibernateDaoTest.java:
```diff
@@ -0,0 +1,171 @@
+
+    @BeforeEach
+    void setUp() {
+        closeable = MockitoAnnotations.openMocks(this);
```
Don't forget to add the `close()`statement.
```java
@AfterEach
void tearDown() {
    closeable.close()
}
```

Francois389 commented on src/test/java/sae/semestre/six/generic/DaoHibernateAbstraitTest.java:
```diff
@@ -0,0 +1,170 @@
+
+    @BeforeEach
+    void setUp() {
+        closeable = MockitoAnnotations.openMocks(this);
```
Don't forget to add the `close()`statement.
```java
@AfterEach
void tearDown() {
    closeable.close()
}
```

Commit:
test: add teardown methods for resource cleanup

Introduces @AfterEach teardown methods in test classes to ensure
properly closing of resources after each test execution. Updates
comments for clarity and consistency in English.

Francois389 reviewed (approved):
👍
